File: Solutions/Medium/3634. Minimum Removals to Balance Array/Solution.cs
public class Solution {
    public int MinRemoval(int[] nums, int k) {
        int n = nums.Length, right = 0, result = Int32.MaxValue;
        Array.Sort(nums);
        for(int i = 0; i < n; i++){
            long max = (long)nums[i] * k;
            while(right < n && nums[right] <= max){
                right++;
            }
            result = Math.Min(n - (right - i), result);
        }
        return result;
    }
}

File: Solutions/Medium/3634. Minimum Removals to Balance Array/Solution.py
class Solution:
    def minRemoval(self, nums: List[int], k: int) -> int:
        n: int = len(nums)
        nums.sort()
        result = float('inf')
        right: int = 0
        for i in range(0, n):
            while(right < n and nums[i] * k >= nums[right]):
                right += 1
            result = min(n - (right - i), result)
        return result